usr/share/rear/prep/GNU/Linux/400_guess_kernel.sh

```diff
@@ -0,0 +1,98 @@
+# 400_guess_kernel.sh
+#
+# Guess kernel if not set yet or error out,
+# for diverse Architectures (arm, aarch64, etc.)
+#
+# This file is part of Relax-and-Recover, licensed under the GNU General
+# Public License. Refer to the included COPYING for full text of license.
+
+# When KERNEL_FILE is specified use that:
+if test -s "$KERNEL_FILE" ; then
```

Comment:
If I specify (e.g. in /etc/rear/local.conf):
```
...
KERNEL_FILE="/my_made-up_kernel"
...
```

Shouldn't we stop here with ERROR and throw information that _"my_made-up_kernel"_ doesn't exist (or is not a real kernel), instead of silently override user set value ?

V.

Reply:
I use now in 400_guess_kernel.sh
<pre>
# When KERNEL_FILE is specified by the user use that
# (KERNEL_FILE is empty in default.conf):
if test "$KERNEL_FILE" ; then
    LogPrint "Using specified KERNEL_FILE '$KERNEL_FILE' as kernel in the recovery system"
    return
fi
</pre>
because what the user has specified must be used "as is" by ReaR
and not silently changed.
Because now it also errors out in various output stage scripts
when kernel or initrd are missing it behaves sufficiently fail-safe
when the user has specified a non-existent KERNEL_FILE

E.g. when I have in my etc/rear/local.conf
<pre>
KERNEL_FILE="/qqq/QQQ"
</pre>
it errors out as in
https://github.com/rear/rear/pull/1985#issuecomment-442498812
so that the user should be sufficiently safe against accidental
typos in what he specified, e.g. for
<pre>
KERNEL_FILE="/boot/vmlinus-4.12.14-lp150.12.25-default"
</pre>
it errors out.

In contrast when I have my etc/rear/local.conf
<pre>
KERNEL_FILE="/etc/os-release"
</pre>
it does not error out because in general when the user
has specified something ReaR must obey.

I think it is over the top (at least how ReaR is currently meant)
to do detailed checks or analysis whether or not what the
user has specified will actually work.

Reply:
I enhanced prep/GNU/Linux/400_guess_kernel.sh
how it behaves when KERNEL_FILE is specified:

It uses the symlink target of a specified KERNEL_FILE
(if the specified KERNEL_FILE is a symlink)
to match what the autodetection code does and
it errors out now directly in prep/GNU/Linux/400_guess_kernel.sh
when the specified KERNEL_FILE does not exist or is a broken symlink
to better match "Try to care about possible errors" in
https://github.com/rear/rear/wiki/Coding-Style

Reply:
I further enhanced prep/GNU/Linux/400_guess_kernel.sh

Now it errors out when for an autodetected file the `file` command
output does not contain the word 'kernel' (ingnore case) and it
shows a WARNING in this case when KERNEL_FILE was specified
by the user i.e. the user can enforce to use a file as kernel where
the `file` command output does not contain the word kernel.

@gozora 
thanks for your hint to use the `file` command to verify if a file
seems to be actually a kernel.

I think it is o.k. to error out in 'prep' stage (that only runs during
"rear mkrescue/mkbackup") when autodetected things look bad
because the user can enforce it by manually specifying KERNEL_FILE
and in general better error out during "rear mkrescue/mkbackup"
than havin a useless recovery system (i.e. better safe than sorry).